qupulse/hardware/awgs/decadac.py

```diff
@@ -0,0 +1,440 @@
+
+
+    def get_ascii(self):
+        command = 'B' + str(self._channel[0]) + ';C' + str(self._channel[1]) + ';'
+        command += 'D' + str(self._voltage) + ';'
+        return command
+
+    def is_ramp(self):
+        return False
+
+
+class Ramp:
+    def __init__(self, duration: int, voltage_step, channel: tuple, memory_dict: dict):
+        self._step = voltage_step
+        self._channel = channel # in the form of (B,C) where B is the board and C is the channel. Possible values: B 0-4, C 0-3
+        self._memory_dict = memory_dict # dictionary correlating the channel to the location in memory for the channel voltage
+        self._duration = duration
+
+    def get_ascii(self):
+        command = 'A' + str(self._memory_dict[self._channel]) + ';'
+        command += '+' + str(self._step) + ';'
+        return command
+
+    def is_ramp(self):
+        return True
+
+    @property
+    def duration(self):
+        return self._duration
+
+
+class Single:
+
+    def __init__(self, current_label: int, duration: int):
+        self._instructions = []
+        self._current_label = current_label
+        self._duration = duration
+
+    def add_instruction(self, instruction):
+        self._instructions.append(instruction)
+
+    def get_ascii(self):
+        command = '*' + str(self._current_label) + ':'
+        command += 'X' + str(768+self._current_label) + ';'
+        for instruction in self._instructions:
+            if instruction.is_ramp():
+                command += instruction.get_ascii()
+        command += '$' + str(self._duration)
+        return command, self._current_label+1
+
+
+class Iteration:
+
+    def __init__(self, current_label: int, duration: int, loops: int, memory_location: int):
+        self._instructions = []
+        self._current_label = current_label
+        self._duration = duration
+        self._memory_location = memory_location
+        self._loops = loops
+
+    def get_ascii(self):
+        command = 'A' + str(self._memory_location) + 'P' + str(self._loops)
+        command += '*' + str(self._current_label) + ':'
+        command += 'X' + str(768 + self._current_label) + ';'
+        for instruction in self._instructions:
+            command += instruction.get_ascii()
+        command += '$' + str(self._duration)
+        command += 'A' + str(self._memory_location)
+        command += 'X' + str(1280 + self._current_label) + ';'
+        return command, self._current_label+1
+
+    def add_instruction(self, instruction):
+        self._instructions.append(instruction)
+
+
+class DecaDACAWG(AWG):
+
+    def __init__(self, port: str, identifier='DecaDAC',):
+        super().__init__(identifier=identifier)
+
+        if port is not str:
+            raise TypeError(
+                'Declared port must be a string. Usually in the form of COM + n, with n being the port number')
+        self._ser = serial.Serial(port, 9600, timeout=0)
+        self._channel_voltage_location = {1: 1545,

```

Comment:
Put this information in a global constant with an all capital name. And it looks like the formula is

```python
VOLT_LOCATION_OFFSET = 1545
def get_channel_voltage_location(channel: int) -> int:
    return VOLT_LOCATION_OFFSET + channel * 6
```